### PR TITLE
Add two network checks

### PR DESCRIPTION
**Pull request by tsmithv11**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Created a construct for checking for any port open to the world (0.0.0.0/0 or empty)

CKV_ALI_2: Ensure security group does not allow internet traffic to SSH port (22)

CKV_ALI_3: Ensure security group does not allow internet traffic to RDP port (3389)

### CODE REVIEW

File: checkov/terraform/checks/resource/alicloud/AbsSecurityGroupUnrestrictedIngress.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                self.evaluated_keys = ['port_range', 'cidr_ip']
+                if self.contains_violation(conf):
+                    return CheckResult.FAILED
+                return CheckResult.PASSED
+            return CheckResult.UNKNOWN
+
+        return CheckResult.PASSED
+
+    def contains_violation(self, conf):
+        from_port = force_int(force_list(conf.get('port_range',[{-1}]))[0].split('/')[0])
```
Comment (gruebel) [inline checkov/terraform/checks/resource/alicloud/AbsSecurityGroupUnrestrictedIngress.py:45]: i think I would recommend to to check if there is a `port_range` argument, if not then directly return `PASSED`. Then you don't have to use the default here.

File: checkov/terraform/checks/resource/alicloud/AbsSecurityGroupUnrestrictedIngress.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        to_port = force_int(force_list(conf.get('port_range',[{-1}]))[0].split('/')[1])
+
+        if from_port == 0 and to_port == 0:
+            to_port = 65535
+
+        if from_port is not None and to_port is not None and (from_port <= self.port <= to_port):
```
Comment (gruebel) [inline checkov/terraform/checks/resource/alicloud/AbsSecurityGroupUnrestrictedIngress.py:51]: from and to port should never be `None` if you add my suggestion about checking the `port_range` key.

**Review by gruebel — COMMENTED** [review]
nice 🥇 , I added a couple of things to make live easier

File: checkov/terraform/checks/resource/alicloud/AbsSecurityGroupUnrestrictedIngress.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                return CheckResult.PASSED
+            return CheckResult.UNKNOWN
+
+        return CheckResult.PASSED
+
+    def contains_violation(self, conf):
+        from_port = force_int(force_list(conf.get('port_range',[{-1}]))[0].split('/')[0])
+        to_port = force_int(force_list(conf.get('port_range',[{-1}]))[0].split('/')[1])
+
+        if from_port == 0 and to_port == 0:
```
[the remark this answers is not in the document]
Comment (tsmithv11) [reply checkov/terraform/checks/resource/alicloud/AbsSecurityGroupUnrestrictedIngress.py]: Removed

File: checkov/terraform/checks/resource/alicloud/AbsSecurityGroupUnrestrictedIngress.py (hunk 1 above)
Comment (tsmithv11) [reply checkov/terraform/checks/resource/alicloud/AbsSecurityGroupUnrestrictedIngress.py:47]: Added

**Comment by tsmithv11** [comment]
> nice 🥇 , I added a couple of things to make live easier

Thank you for the feedback!

**Pull request merged**